3rdparty/openssl/CMakeLists.txt

```diff
@@ -46,6 +44,23 @@ if(WIN32)
 		VERBATIM
 	)
 	add_custom_target(openssl_build DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/build/lib/libeay32.lib" "${CMAKE_CURRENT_BINARY_DIR}/build/lib/ssleay32.lib")
+elseif(ANDROID) # cross-compile
+	find_package(Perl REQUIRED)
+
+	set(ENV{ANDROID_NDK_ROOT} "$ENV{ANDROID_NDK}")
+	set(ENV{PATH} "$ENV{ANDROID_NDK_ROOT}/toolchains/llvm/prebuilt/linux-x86_64/bin:$ENV{ANDROID_NDK_ROOT}/toolchains/x86_64-4.9/prebuilt/linux-86_64/bin:$ENV{ANDROID_NDK_ROOT}")
+	add_custom_command(
+		OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/build/lib/libcrypto.a" "${CMAKE_CURRENT_BINARY_DIR}/build/lib/libssl.a"
+		COMMAND make clean
+		# not sure if linux-generic64 is right (https://stackoverflow.com/questions/26779531/openssl-for-android-x86-64-and-mips64)
+		# but android-x86_64 is not even recognized (should be according to this https://github.com/openssl/openssl/blob/master/NOTES-ANDROID.md)
+		COMMAND "${PERL_EXECUTABLE}" "${CMAKE_CURRENT_SOURCE_DIR}/openssl/Configure" linux-generic64 no-asm no-shared -D__ANDROID_API__=${ANDROID_API_LEVEL} --openssldir=${CMAKE_CURRENT_BINARY_DIR}/build
+		COMMAND make -j64
+		COMMAND make install
+		WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/openssl"
+		VERBATIM
+	)
+	add_custom_target(openssl_build DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/build/lib/libcrypto.a" "${CMAKE_CURRENT_BINARY_DIR}/build/lib/libssl.a")
```

Comment:
@AnastaZIuk I'll let you fix this when tackling the "out of source build" for OpenSSL so our module isn't perpetually dirty.

Reply:
@AnastaZIuk is it fixed in the latest `criss_android`?

Reply:
I think it isn't, OpenSSL builds in dirty 3rdparty that may lead to troubles when using various toolchains assuming same directory, also we do not provide building OpenSSL on windows when cross-compiling

Reply:
@AnastaZIuk any news about making OpenSSL build out of source?

Reply:
not yet, I've been doing different things

Reply:
@AnastaZIuk can we have OpenSSL and zlib build out of source properly?

The submodules always show up as dirty which is incredibly annoying when working with BaW (because it also makes Nabla dirty)

Reply:
yes, priority now?

Reply:
would be nice to do while you're doing build systems right now